src/wally-connector.ts

```diff
@@ -234,96 +243,168 @@ class WallyConnector {
     localStorage.removeItem(`wally:${this.clientId}:state:token`);
   }
 
-  async request(req: RequestObj): Promise<any> {
+  /**
+   * This is the major exposed method for supporting JSON RPC methods
+   * and associated wallet/blockchain functionality.
+   * There are two main types of requests: those that require wallet info
+   * (address, signing), and those that do not (gas prices, last block).
+   * We route the former to customized endpoints on the backend that handle
+   * this extra wallet fetching and logic, and the latter to an endpoint
+   * that essentially works as a passthrough to ethers/alchemy.
+   *
+   * TODO: Move requesting logic and helpers to separate file/module
+   * @param req
+   * @param req.method - the name of the RPC method
+   * @param req.params - the required parameters for the method
+   * @returns Promise<MethodResponse> | null
+   * @see https://ethereum.org/en/developers/docs/apis/json-rpc/#json-rpc-methods
+   */
+  async request<T extends MethodNameType>(
+    req: RequestObj<T>
+  ): Promise<MethodResponse<T> | null> {
     if (!this.isLoggedIn()) {
       await this.loginWithEmail();
     }
 
-    switch (req.method) {
-      case MethodName.REQUEST_ACCOUNTS:
-        return this.requestAccounts();
-      // TODO: figure out which name to use
-      case MethodName.PERSONAL_SIGN:
-      case MethodName.SIGN:
-        return this.signMessage(req.params);
-      case MethodName.GET_BALANCE:
-        return this._request(req.method, req.params);
+    if (Object.values(WallyMethodName).indexOf(req.method as any) > -1) {
```

Comment:
could we make this into a helper function just to make the code easier to read?

Reply:
for sure 👍  - funny you mention it, I was actually trying to get a type guard function setup here earlier and gave up haha - i'll get it figured